app/src/main/java/com/example/footballmanager/ui/theme/navigation/BottomNavigationBar.kt

```diff
@@ -0,0 +1,104 @@
+fun ButtonNavigationBar(
+    modifier: Modifier = Modifier,
+    onNavigateToScreen: (ScreensEnum) -> Unit
+) {
+
+    val masterViewModel: MasterViewModel = hiltViewModel()
+
+    Column(
+        verticalArrangement = Arrangement.spacedBy(dimensionResource(id = R.dimen.bot_nav_bar_items_spacing), Alignment.Top),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = modifier
+            .windowInsetsPadding(
+                WindowInsets.safeDrawing.only(WindowInsetsSides.Horizontal + WindowInsetsSides.Bottom)
+            )
+            .fillMaxWidth()
+            .background(color = colorResource(id = R.color.highlighted_element_color))
+            .padding(top = dimensionResource(id = R.dimen.bot_nav_bar_padding))
+    ) {
+        Row(
+            horizontalArrangement = Arrangement.SpaceBetween,
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(horizontal = dimensionResource(id = R.dimen.bot_nav_bar_row_padding))
+                .background(colorResource(R.color.highlighted_element_color))
+        ) {
+            val buttonColor = NavigationBarItemColors(
+                selectedIconColor = colorResource(id = R.color.app_red_motive),
+                selectedTextColor = Color.White,
+                selectedIndicatorColor = colorResource(R.color.highlighted_element_color),
+                unselectedIconColor = colorResource(id = R.color.unselected_navigation_item),
+                unselectedTextColor = colorResource(id = R.color.unselected_navigation_item),
+                disabledIconColor = colorResource(R.color.highlighted_element_color),
+                disabledTextColor = colorResource(R.color.highlighted_element_color)
+            )
+            NavigationBar(
+                modifier = Modifier
+                    .background(colorResource(R.color.highlighted_element_color)),
+                containerColor = colorResource(R.color.highlighted_element_color),
+            ) {
+                ScreensEnum.entries.forEach { screen ->
+                    NavigationBarItem(
+                        selected = masterViewModel.currentBotNavSelection == screen,
+                        onClick = { onNavigateToScreen(screen) },
+                        icon = { NavItemIcon(imgVec = ImageVector.vectorResource(id = screen.iconRes)) },
+                        label = { NavItemText(screen.name) },
+                        colors = buttonColor,
+                        modifier = Modifier
+                            .background(colorResource(R.color.highlighted_element_color))
+                    )
+                }
+            }
+
+        }
+    }
+}
+
+@Composable
+fun NavItemIcon(modifier: Modifier = Modifier, imgVec: ImageVector) {
+    Icon(
+        imageVector = imgVec,
+        modifier = modifier.size(dimensionResource(id = R.dimen.bot_nav_bar_item_size)),
+        contentDescription = null
+    )
+}
+
+@Composable
+fun NavItemText(name: String) {
```

Comment:
You don't need to isolate each component in a separate composable, especially if they are as simple as this Text